devito/finite_differences/coefficients.py

```diff
@@ -50,13 +50,17 @@ def __init__(self, deriv_order, function, dimension, weights):
         self._check_input(deriv_order, function, dimension, weights)
 
         # Ensure the given set of weights is the correct length
+        try:
+            wl = weights.shape[-1]-1
+        except AttributeError:
+            wl = len(weights)-1
         if dimension.is_Time:
-            if len(weights)-1 != function.time_order:
-                raise ValueError("Number FD weights provided does not "
+            if wl != function.time_order:
+                raise ValueError("Number of FD weights provided does not "
                                  "match the functions space_order")
         elif dimension.is_Space:
-            if len(weights)-1 != function.space_order:
-                raise ValueError("Number FD weights provided does not "
+            if wl != function.space_order:
+                raise ValueError("Number of FD weights provided does not "
```

Comment:
For future PR, but this still asumes the use knows what the indices are with the ordering, because for example for staggered FD in my new PR the indices are not ordered anymore in some cases. Mayeb would be good to give it as a dict `pos: w` or something like that.

Reply:
Yeah, good point will add note.